pkgs/sketch_pad/lib/main.dart

```diff
@@ -271,215 +284,129 @@ class _DartPadMainPageState extends State<DartPadMainPage> {
   @override
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
-    final scaffold = Scaffold(
-      appBar: widget.embedMode
-          ? null
-          : AppBar(
-              backgroundColor: theme.colorScheme.surface,
-              title: SizedBox(
-                height: toolbarItemHeight,
-                child: Row(
-                  children: [
-                    dartLogo(width: 32),
-                    const SizedBox(width: denseSpacing),
-                    Text(appName,
-                        style: TextStyle(
-                            color: Theme.of(context).colorScheme.onSurface)),
-                    const SizedBox(width: defaultSpacing * 4),
-                    NewSnippetWidget(appServices: appServices),
-                    const SizedBox(width: denseSpacing),
-                    const ListSamplesWidget(),
-                    const SizedBox(width: defaultSpacing),
-                    // title widget
-                    Expanded(
-                      child: Center(
-                        child: ValueListenableBuilder<String>(
-                          valueListenable: appModel.title,
-                          builder: (_, String value, __) => Text(value),
-                        ),
-                      ),
-                    ),
-                    const SizedBox(width: defaultSpacing),
+    final executionWidget = ExecutionWidget(
+      appServices: appServices,
+      appModel: appModel,
+      key: _executionWidgetKey,
+    );
+    final loadingOverlay = LoadingOverlay(
```

Comment:
+1 - great refactoring (here and `EditorWithButtons`); it helps us reason about the responsive layout.